src/canvas-view/canvas/canvas-directive.js

```diff
@@ -0,0 +1,440 @@
+(function () {
+  'use strict';
+
+  angular.module('patternfly.canvas')
+  .filter('trustAsResourceUrl', ['$sce', function ($sce) {
+    return function (val) {
+      return $sce.trustAsResourceUrl(val);
+    };
+  }])
+
+  //
+  // Directive that generates the rendered chart from the data model.
+  //
+  .directive('pfCanvas', function ($document) {
+    return {
```

Comment:
OK, I would have preferred not to introduce directives but I guess this could be refactored at a later time.